x-pack/plugins/timelines/server/search_strategy/timeline/factory/events/all/query.events_all.dsl.ts

```diff
@@ -58,7 +58,7 @@ export const buildTimelineEventsAllQuery = ({
       return {
         [field]: {
           order: item.direction,
-          unmapped_type: item.type,
+          unmapped_type: item.type === 'date' ? 'date' : 'keyword',
```

Comment:
A longer-term fix that uses the `esTypes` metadata here, and in the column tooltips would be a more most robust fix, but in the short term, perhaps only `item.type` `string` should be special cased, because

```ts
unmapped_type: item.type === 'date' ? 'date' : 'keyword',
```

returns unexpected results for other types.

Example: sorting by the `source.ip` field in `main` returns ` "unmapped_type": "ip"` in the following DSL:

```json
"sort": [
    {
      "source.ip": {
        "order": "asc",
        "unmapped_type": "ip"
      }
    }
  ],
```

but with the proposal above, returns ` "unmapped_type": "keyword"`:

```json
  "sort": [
    {
      "source.ip": {
        "order": "asc",
        "unmapped_type": "keyword"
      }
    }
  ],
```

For a short-term fix (that doesn't use the `esTypes` metadata), consider something like:

```ts
unmapped_type: item.type === 'string' ? 'keyword' : item.type,
```

I desk tested the above with a multi-column sort on the following fields:

- `@timestamp`
- `source.ip`
- `user.name`

as shown in the screenshot below:

![multi_type_sort](https://user-images.githubusercontent.com/4459398/163597754-291f08cc-76e3-4e79-a0de-542d9f3d469b.png)

and the request contained the expected types for all the fields:

```json
"sort": [
    {
      "@timestamp": {
        "order": "desc",
        "unmapped_type": "date"
      }
    },
    {
      "source.ip": {
        "order": "desc",
        "unmapped_type": "ip"
      }
    },
    {
      "user.name": {
        "order": "desc",
        "unmapped_type": "keyword"
      }
    }
  ],
```

If that makes sense, consider creating a unit test where the input is a multi-column sort on `@timestamp`, `source.ip`, and `user.name` to assert that types returned are `date`, `ip`, and `keyword` respectively.